src/dlgConnectionProfiles.cpp

```diff
@@ -1662,6 +1676,156 @@ void dlgConnectionProfiles::fillout_form()
     updateDiscordStatus();
 }
 
+void dlgConnectionProfiles::setProfileIcon(const QFont& font) const
+{
+    for (int i = 0; i < mProfileList.size(); i++) {
+        const QString& profileName = mProfileList.at(i);
+        if (profileName.isEmpty()) {
+            continue;
+        }
+
+        if (hasCustomIcon(profileName)) {
+            loadProfileIcon(font, profileName);
+        } else {
+            // mProfileList is derived from a filesystem directory, but MacOS is not
+            // necessarily case preserving for file names so any tests on them
+            // should be case insensitive
+            // skip creating icons for default MUDs as they are already created above
+            if (mDefaultGames.contains(profileName, Qt::CaseInsensitive)) {
+                continue;
+            }
+
+            generateProfileIcon(font, i, profileName);
+        }
+    }
+}
+
+bool dlgConnectionProfiles::hasCustomIcon(const QString& profileName) const
+{
+    return QFileInfo::exists(mudlet::getMudletPath(mudlet::profileDataItemPath, profileName, QStringLiteral("profileicon")));
+}
+
+void dlgConnectionProfiles::loadProfileIcon(const QFont& font, const QString& profileName) const
+{
+    auto pItem = new QListWidgetItem(profileName);
+    pItem->setFont(font);
+    pItem->setForeground(QColor(Qt::white));
+    auto profileIconPath = mudlet::getMudletPath(mudlet::profileDataItemPath, profileName, QStringLiteral("profileicon"));
```

Comment:
Can we have this block (until 1715) factored out somewhere? It repeats in the default server list as well. Maybe even make the default profiles use this function as well?

Reply:
Feedback applied.